lib/sensors/low_power_current.cpp

```diff
@@ -0,0 +1,38 @@
+#include <low_power_current.hpp>
+
+namespace hyped::sensors {
+
+// TODOLater: Confirm failure states for create
+std::optional<LowPowerCurrent> LowPowerCurrent::create(core::ILogger &logger,
+                                                       std::shared_ptr<io::II2c> i2c,
+                                                       const std::uint8_t device_address)
+{
+  return LowPowerCurrent(logger, i2c, device_address);
+}
+
+LowPowerCurrent::LowPowerCurrent(core::ILogger &logger,
+                                 std::shared_ptr<io::II2c> i2c,
+                                 const std::uint8_t device_address)
+    : logger_(logger),
+      i2c_(i2c),
+      device_address_(device_address)
+{
+}
+
+LowPowerCurrent::~LowPowerCurrent()
+{
+}
+
+std::optional<core::Float> LowPowerCurrent::readCurrent()
+{
+  const auto byte = i2c_->readByte(device_address_, kLowPowerCurrentRegister);
+  if (!byte) {
+    logger_.log(core::LogLevel::kFatal, "Failed to read current on channel %d", device_address_);
```

Comment:
channel? same with the comment below. I think "Failed to read low power current" should be fine